### PR TITLE
Improve networking

### PR DESCRIPTION
**Pull request by polarker**

### CODE REVIEW

**Comment by polarker** [comment]
@Lbqds @tdroxler please help me review this PR. This should improve network robustness

File: flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala [hunk 1]
```diff
@@ -83,8 +83,17 @@ trait DiscoveryServerState extends SessionManager {
     }
   }
 
+  // select a number of random peers based on a random clique id
+  def getBootstrapNeighbors(): AVector[BrokerInfo] = {
```
Comment (Lbqds) [inline flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala:87]: It seems that this function is not used.

File: flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala [hunk 2]
```diff
@@ -83,8 +83,17 @@ trait DiscoveryServerState extends SessionManager {
...
+    getNeighbors(selfCliqueId, CliqueId.generate)
+  }
+
   def getNeighbors(target: CliqueId): AVector[BrokerInfo] = {
-    val candidates = AVector.from(table.values.map(_.info).filter(_.cliqueId != target))
+    getNeighbors(target, target)
+  }
+
+  def getNeighbors(filterId: CliqueId, target: CliqueId): AVector[BrokerInfo] = {
+    val candidates = AVector.from(table.values.map(_.info).filter(_.cliqueId != filterId))
```
Comment (tdroxler) [inline flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala:96]: I don't see any difference with the previous implementation, why suddenly it's random?

File: flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala (hunk 1 above)
Comment (polarker) [reply flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala:98]: nice catch!! two more commits pushed

File: flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala (hunk 2 above)
Comment (polarker) [reply flow/src/main/scala/org/alephium/flow/network/DiscoveryServerState.scala:107]: nice catch!! two more commits pushed

**Review by Lbqds — APPROVED** [review]
LGTM

**Pull request merged**